source/content/guides/drupal-9-migration/06-build-tools-to-d9-build-tools.md

```diff
@@ -0,0 +1,242 @@
+---
+
+In this doc, you'll create a new Drupal 9 site and migrate the code from an existing Drupal 8 site to it.
+
+## Will This Guide Work for Your Site?
+
+<Partial file="drupal-9/upgrade-site-requirements.md" />
+
+## Prepare the Local Environment
+
+<Partial file="drupal-9/prepare-local-environment.md" />
+
+This doc uses the following aliases:
+
+- **Alias:** alias
+
+## Build Tools Workflow
+
+This tutorial is for you if you meet the following criterion:
+
+1. You have an external repository outside of pantheon that manages your site's code.
+
+1. You use a build service like Github Actions, Circle CI or Travis to "build" the site.
+
+1. Those build artifacts are pushed to your pantheon repo.
+
+1. Your site has a blue banner and is database-ready for the upgrade to d9
+
+
+## Tools you will need:
+
+brew install terminus jq rsync
+
+
+## Wagons HO!:
+
+* clone your github source repo and cd into the directory, then:
+
+ `git checkout -b d9-upgrade-2021` 
+
+* Let's export the configs from production to make sure our local version has the latest:
+
+```bash
+# Export the latest from the live environment
+# to sites/default/files/config
+terminus drush gk-8.live -- config:export \
+   --destination sites/default/files/config
+
+# Use terminus to get the (read-only) SFTP command
+# specific to the live environment.
+SFTP_COMMAND=$(terminus connection:info gk-8.live \
+   --format=json | jq -r ".sftp_command")
+
+# For rysnc, all we really need is the crazy-long
+# user and host name. Strip out the rest.
+# (Make sure the empty space is there at the end 
+# before the bracket!)
+RSYNC_HOST=${SFTP_COMMAND#sftp -o Port=2222 } 
+
+# Use that hostname to do an rsync from that recent
+# config:export
+rsync -rvlz --copy-unsafe-links --size-only --checksum \
+   --ipv4 --progress -e 'ssh -p 2222' \
+   "${RSYNC_HOST}:files/config" .
+```
+
+   If you do a `git status` it should changed files in the `config` directory if there are any changed configs in production.
+
+### Let's start ugradin' stuff!
+
+```bash
+   composer require \
+	   drupal/upgrade_status:^3 \
+	   drupal/devel:^4 \
+	   drush/drush:^10 \
+	   -W --no-update
+   composer require \
+	   phpunit/phpunit:"^8 | ^9 | ^10" \
+	   phpstan/phpstan \
+		 --dev -W --no-update
+```
+
+* Edit your composer file and change the following:
+
+```json
+  "scripts": {
+    "build-assets": [
+      "@prepare-for-pantheon",
+      "composer install --optimize-autoloader --no-dev"
+    ],
+```
+
+TO:
+
+```json
+[...]
+  "scripts": {
+    "build-assets": [
+      "@prepare-for-pantheon",
+      "composer install --optimize-autoloader"
+    ],
+[...]
+```
+
+   Removing the "--no-dev" portion of that line. This will allow your dev dependencies to be available in the integration environment.
+
+`composer update -W --optimize-autoloader --prefer-dist`
+
+1. If your site doesn't already have a pantheon.yml file, create one ensuring the following values:
```

Comment:
I don't know that we should provide values to use here. php_version might vary by site. drush_version isn't necessary because Build Tools sites should use a site-local Drush. enforce_https full+subdomains might be too much for some sites; if someone copied that without understanding, they just made a one-year mistake.  Let's leave this at just api_version, web_docroot, and database, and tell people to check and confirm their other values.